### PR TITLE
feat: check authorization for GQL owners

### PR DESCRIPTION
**Pull request by JerrySentry**
### Purpose/Motivation

Add a authentication and authorization for a few sensitive fields in the Owners query, we don't want to expose these information to users that don't belong in this org.

### Links to relevant tickets

https://github.com/codecov/engineering-team/issues/1082

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: graphql_api/types/owner/owner.graphql [hunk 1]
```diff
@@ -15,21 +15,21 @@ type Owner {
   repository(name: String!): RepositoryResult!
```
Comment (JerrySentry) [inline graphql_api/types/owner/owner.graphql:15]: These fields can be null now, when the user is not part of the the field will be null and will also have error in the response with the appropriate message.

File: graphql_api/helpers/mutation.py [hunk 2]
```diff
@@ -57,5 +58,20 @@ def authenticated_resolver(instance, info, *args, **kwargs):
     return authenticated_resolver
 
 
+def require_part_of_org(resolver):
+    def authenticated_resolver(queried_owner, info, *args, **kwargs):
+        current_user = info.context["request"].user
+        current_owner = info.context["request"].current_owner
+        if not current_user.is_authenticated or not current_owner:
```
Comment (ajay-sentry) [inline graphql_api/helpers/mutation.py:65]: will current_user always be non-null? Wonder if we should use .get() here instead

File: graphql_api/types/owner/owner.py [hunk 3]
```diff
@@ -81,6 +82,7 @@ def resolve_plan_representation(owner: Owner, info) -> PlanData:
 
 @owner_bindable.field("availablePlans")
 @convert_kwargs_to_snake_case
+@require_part_of_org
```
Comment (ajay-sentry) [inline graphql_api/types/owner/owner.py:85]: what was the determining factor in picking which functions to add the decorator to? I see there are certain functions where we didn't add it so was just curious

File: graphql_api/helpers/mutation.py (hunk 2 above)
Comment (JerrySentry) [reply graphql_api/helpers/mutation.py]: Updated to use .get

File: graphql_api/types/owner/owner.py (hunk 3 above)
Comment (JerrySentry) [reply graphql_api/types/owner/owner.py:87]: https://www.notion.so/sentry/GraphQL-Owner-Type-Sensitive-Fields-f78cb53e77e4428aa6e04b2819d67884
and yes, good catch, added in all the fields

File: graphql_api/helpers/mutation.py [hunk 4]
```diff
@@ -57,5 +58,20 @@ def authenticated_resolver(instance, info, *args, **kwargs):
     return authenticated_resolver
 
 
+def require_part_of_org(resolver):
```
Comment (ajay-sentry) [inline graphql_api/helpers/mutation.py:61]: is it easy to add a UT for this guy?

**Review by ajay-sentry — APPROVED** [review]
just a small comment about UT but otherwise looks good

File: graphql_api/helpers/mutation.py (hunk 4 above)
Comment (JerrySentry) [reply graphql_api/helpers/mutation.py:61]: this function should be fully covered with the test_owner.py tests

**Review by barnett-yuxiang — COMMENTED** [review]
cool

**Pull request merged**